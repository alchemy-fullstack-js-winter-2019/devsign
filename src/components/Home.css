.Home {
  display: grid;
  grid-gap: 10px;
  grid-template-columns: 150px 150px 150px;
  padding: 1 em;
  margin: 20px;
}
header {
  grid-column: 1 / span 4;
  background-color: #84B3DA;
}
footer {
  position: absolute;
  bottom: 0;
}
.sidebar {
  background-color: #84B3DA;
}