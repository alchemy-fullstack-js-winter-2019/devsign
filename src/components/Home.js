import React from 'react';

export default function Home() {
  return (
    <>
      <section>
        <figure>
          <img src="https://pbs.twimg.com/media/C8QsNInXUAAyjZQ.jpg" alt="user avatar" />
          <h3>T_on_A</h3>
        </figure>
      </section>
      <section>
        <label>Post a Tweet</label>
        <input type="text" />
        <button>Tweet</button>
        <button>Add An Image</button>
        <button>Add A Gif</button>
        <button>Add Location</button>
      </section>
      <section>
        <ol>
          <li>List of tweets from users you are following</li>
        </ol>
      </section>
      <aside>
        <ul>
          <section>
            <figure>
              <img src="https://pbs.twimg.com/media/C8QsNInXUAAyjZQ.jpg" alt="user avatar" />
            </figure>
            <h3>Jill Hill</h3>
            <p>Todays lab sucks</p>
          </section>
          <section>
            <figure>
              <img src="https://pbs.twimg.com/media/C8QsNInXUAAyjZQ.jpg" alt="user avatar" />
            </figure>
            <h3>Jack Hill</h3>
            <p>Todays lab is AMAZING</p>
          </section>
          <section>
            <figure>
              <img src="https://pbs.twimg.com/media/C8QsNInXUAAyjZQ.jpg" alt="user avatar" />
            </figure>
            <h3>Hilly Jill</h3>
            <p>Todays lab is all uphill</p>
          </section>
        </ul>
      </aside>
    </>
  );
}
