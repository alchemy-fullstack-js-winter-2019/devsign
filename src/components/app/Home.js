import React, { Fragment } from 'react';
import Header from './Header';
import Trending from '../chirps/Trending';
import ChirpsContainer from '../../containers/PopularChirpsContainer';
import MiniProfileContainer from '../../containers/MiniProfileContainer';
import { Div1, Div2, Section } from './HomeStyles';

export default function DesktopHome() {
  return (
    <Fragment>
      <Header />
      <Section>
        <Div1>
          <MiniProfileContainer />
          <Trending />
        </Div1>
        <Div2>
          <ChirpsContainer />
        </Div2>
      </Section>
    </Fragment>
  );
}
