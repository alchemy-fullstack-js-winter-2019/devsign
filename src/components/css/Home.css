.feed {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 20px;

  h2 {
    color: red;
  }
}
