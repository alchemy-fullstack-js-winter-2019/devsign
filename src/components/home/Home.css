
img {
  max-height: 20vh;
  border-radius: 60px
}
ul {
  padding: 0;
}
.profile  {
  display: none;
  img {
    height: 10vh;
  }
  h3 {
    font-size: 1em;
  }
}

.post {
  margin: 2px;
  background-color: #DEEBED;
  button {
    background-color: #D8D8D8;
  }
  input {
    border: 1px solid #D8D8D8;
  }
}

