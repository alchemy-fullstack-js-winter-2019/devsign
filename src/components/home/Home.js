import React from 'react';

const Home = () => {
  return (
    < >

    {/* MOBILE MOCKUP */}
      <main>
        <section id="all-tweets">
          <article style={{ padding: '.8em' }}>
            <img src='../assets/circle_sm.png' 
              style={{ float: 'left', padding: '.1em .5em 0 0' }}
              alt="profile-pic"
            />
            <h3 style={{ display: 'inline' }}>Aasdfa</h3>
            <br/>
            <span>
            Lorem ipsum dolor sit amet, uioua adipiscing elit, sed do eiusmod. #asdfasdf #tehklj
            </span>
          </article>
          <article style={{ padding: '.8em' }}>
            <img src='../assets/circle_sm.png' 
              style={{ float: 'left', padding: '.1em .5em 0 0' }}
              alt="profile-pic"
            />
            <h3 style={{ display: 'inline' }}>Tio</h3>
            <br/>
            <span>
            Ut enim ad minim veniam, @jlakdsf quis nostrud #lkjasfd
            </span>
          </article>
          <article style={{ padding: '.8em' }}>
            <img src='../assets/circle_sm.png' 
              style={{ float: 'left', padding: '.1em .5em 0 0' }}
              alt="profile-pic"
            />
            <h3 style={{ display: 'inline' }}>SssSss</h3>
            <br/>
            <span>
            Duis aute irure dolor in @reprehenderit in voluptate velit esse #cillum #lkjlkjlkj
            </span>
          </article>
          <article style={{ padding: '.8em' }}>
            <img src='../assets/circle_sm.png' 
              style={{ float: 'left', padding: '.1em .5em 0 0' }}
              alt="profile-pic"
            />
            <h3 style={{ display: 'inline' }}>Zzzelkj</h3>
            <br/>
            <span>
            Excepteur sint occaecat cupidatat, 90 - 69 qui officia deserunt @asdf #asdfasdf
            </span>
          </article>
          <article style={{ padding: '.8em' }}>
            <img src='../assets/circle_sm.png' 
              style={{ float: 'left', padding: '.1em .5em 0 0' }}
              alt="profile-pic"
            />
            <h3 style={{ display: 'inline' }}>Ulkj78</h3>
            <br/>
            <span>
            Consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore jklej oiugj!
            </span>
          </article>
        </section>
      </main>

    </>
  );
};

export default Home;
