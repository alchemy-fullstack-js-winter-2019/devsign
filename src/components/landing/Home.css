.main {
  ul {
    padding: 0;
  }
  .profile  {
    display: none;
    img {
      height: 10vh;
    }
    h3 {
      font-size: 1em;
    }
    img {
      max-height: 20vh;
      border-radius: 60px
    }
  }
  
  .post {
    margin: 2px;
    background-color: #DEEBED;
    button {
      background-color: #D8D8D8;
    }
    input {
      border: 1px solid #D8D8D8;
    }
  }
  
  @media screen and (min-width 1000px) {
    .profile {
      
    }
  }
}  
