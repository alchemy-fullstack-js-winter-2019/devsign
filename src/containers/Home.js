import { connect } from 'react-redux';
import Home from '../components/home/Home';
import { getTweetsActions } from '../actions/Tweets';
import { getAllTweets } from '../selectors/Tweets';



const mapStateToProps = state => ({
    tweets: getAllTweets(state)
});

const mapDispatchToProps = dispatch => ({
    fetchTweets() {
        dispatch(getTweetsActions());
    }
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(Home);
