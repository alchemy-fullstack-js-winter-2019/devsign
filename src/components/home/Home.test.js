import React from 'react';
import { shallow } from 'enzyme';
import Home from './Home';

describe('Home', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(
      <Home/>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
