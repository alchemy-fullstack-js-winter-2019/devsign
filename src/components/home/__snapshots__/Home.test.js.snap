// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches a snapshot 1`] = `
<Fragment>
  <main>
    <section
      id="all-tweets"
    >
      <article
        style={
          Object {
            "padding": ".8em",
          }
        }
      >
        <img
          alt="profile-pic"
          src="../assets/circle_sm.png"
          style={
            Object {
              "float": "left",
              "padding": ".1em .5em 0 0",
            }
          }
        />
        <h3
          style={
            Object {
              "display": "inline",
            }
          }
        >
          Aasdfa
        </h3>
        <br />
        <span>
          Lorem ipsum dolor sit amet, uioua adipiscing elit, sed do eiusmod. #asdfasdf #tehklj
        </span>
      </article>
      <article
        style={
          Object {
            "padding": ".8em",
          }
        }
      >
        <img
          alt="profile-pic"
          src="../assets/circle_sm.png"
          style={
            Object {
              "float": "left",
              "padding": ".1em .5em 0 0",
            }
          }
        />
        <h3
          style={
            Object {
              "display": "inline",
            }
          }
        >
          Tio
        </h3>
        <br />
        <span>
          Ut enim ad minim veniam, @jlakdsf quis nostrud #lkjasfd
        </span>
      </article>
      <article
        style={
          Object {
            "padding": ".8em",
          }
        }
      >
        <img
          alt="profile-pic"
          src="../assets/circle_sm.png"
          style={
            Object {
              "float": "left",
              "padding": ".1em .5em 0 0",
            }
          }
        />
        <h3
          style={
            Object {
              "display": "inline",
            }
          }
        >
          SssSss
        </h3>
        <br />
        <span>
          Duis aute irure dolor in @reprehenderit in voluptate velit esse #cillum #lkjlkjlkj
        </span>
      </article>
      <article
        style={
          Object {
            "padding": ".8em",
          }
        }
      >
        <img
          alt="profile-pic"
          src="../assets/circle_sm.png"
          style={
            Object {
              "float": "left",
              "padding": ".1em .5em 0 0",
            }
          }
        />
        <h3
          style={
            Object {
              "display": "inline",
            }
          }
        >
          Zzzelkj
        </h3>
        <br />
        <span>
          Excepteur sint occaecat cupidatat, 90 - 69 qui officia deserunt @asdf #asdfasdf
        </span>
      </article>
      <article
        style={
          Object {
            "padding": ".8em",
          }
        }
      >
        <img
          alt="profile-pic"
          src="../assets/circle_sm.png"
          style={
            Object {
              "float": "left",
              "padding": ".1em .5em 0 0",
            }
          }
        />
        <h3
          style={
            Object {
              "display": "inline",
            }
          }
        >
          Ulkj78
        </h3>
        <br />
        <span>
          Consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore jklej oiugj!
        </span>
      </article>
    </section>
  </main>
</Fragment>
`;
